### PR TITLE
Adding big5 `coalesce` queries

### PR DESCRIPTION
**Pull request by aalva500-prog**
### Description
Adding big5 queries for the new `coalesce` function enhancements.

### Related Issues
https://github.com/opensearch-project/sql/issues/4005

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
 - [ ] New functionality has javadoc added.
 - [ ] New functionality has a user manual doc added.
- [ ] New PPL command [checklist](https://github.com/opensearch-project/sql/blob/main/DEVELOPER_GUIDE.rst#new-ppl-command-checklist) all confirmed.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff` or `-s`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: integ-test/src/test/resources/big5/queries/coalesce_empty_string_priority.ppl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+source = big5
```
Comment (LantaoJin) [inline integ-test/src/test/resources/big5/queries/coalesce_empty_string_priority.ppl]: What's purpose to add the ppl files? Please update the `PPLBig5IT`.

Comment (aalva500-prog) [reply integ-test/src/test/resources/big5/queries/coalesce_empty_string_priority.ppl]: The purpose of adding the ppl files is to have the source query for new features/enhancements in commands/functions that include benchmark. We are testing performance in big5 data set whenever we improve an existing command/function or create a new one. Hope that makes sense.

Comment (aalva500-prog) [reply integ-test/src/test/resources/big5/queries/coalesce_empty_string_priority.ppl]: Hi @LantaoJin, I tried to add the queries to the `PPLBig5IT` test suite as recommended. During the integration test execution, I have identified two issues that require clarification:

**Issue 1: Calcite Configuration in PPLBig5IT**
The `PPLBig5IT` class disables Calcite by default in its initialization method. If the new command or enhancements are designed to function exclusively with Calcite enabled, the tests will fail in this configuration.

```java
@Override
public void init() throws Exception {
    super.init();
    loadIndex(Index.BIG5);
    disableCalcite();
}
```

**Issue 2: Missing Field in CalcitePPLBig5IT**
When executing tests through CalcitePPLBig5IT, I encounter a field resolution error for `host.name`. The error indicates this field is not present in the current dataset, though it appears to be part of the Big5 dataset mappings used in our EC2 performance testing environment.

```java
org.opensearch.client.ResponseException: method [POST], host [http://[::1]:61814], URI [/_plugins/_ppl], status line [HTTP/1.1 400 Bad Request]
{
  "error": {
    "reason": "Invalid Query",
    "details": "field [host.name] not found; input fields are: [agent, agent.ephemeral_id, agent.id, agent.name, agent.type, agent.version, process, process.name, log, log.file, log.file.path, message, tags, cloud, cloud.region, input, input.type, @timestamp, ecs, ecs.version, data_stream, data_stream.dataset, data_stream.namespace, data_stream.type, meta, meta.file, host, metrics, metrics.size, metrics.tmin, aws, aws.cloudwatch, aws.cloudwatch.ingestion_time, aws.cloudwatch.log_group, aws.cloudwatch.log_stream, event, event.dataset, event.id, event.ingested, _id, _index, _score, _maxscore, _sort, _routing]",
    "type": "IllegalArgumentException"
  },
  "status": 400
}
```

Could you please confirm whether the `host.name` field should be included in the `Big5` dataset used for integration testing in this project? This will help determine if we need to update the dataset or modify the test queries accordingly.

Thank you for your guidance on resolving these issues.

File: integ-test/src/test/java/org/opensearch/sql/calcite/big5/PPLBig5IT.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 @FixMethodOrder(MethodSorters.JVM)
 public class PPLBig5IT extends PPLIntegTestCase {
```
Comment (ykmr1224) [inline integ-test/src/test/java/org/opensearch/sql/calcite/big5/PPLBig5IT.java:19]: It can be a separate task, but we might want to scan the files under `big5/queries` and automatically execute all.

**Review by ykmr1224 — APPROVED** [review]

File: integ-test/src/test/java/org/opensearch/sql/calcite/big5/PPLBig5IT.java (hunk 2 above)
Comment (aalva500-prog) [reply integ-test/src/test/java/org/opensearch/sql/calcite/big5/PPLBig5IT.java:19]: Yeah, that's a great idea.

File: integ-test/src/test/resources/big5/queries/coalesce_empty_string_priority.ppl [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+| eval result = coalesce('', `metrics.size`)
```
Comment (noCharger) [inline integ-test/src/test/resources/big5/queries/coalesce_empty_string_priority.ppl]: This appears to be an edge case query. Any implementation nuances you'd like to benchmark?

File: integ-test/src/test/resources/big5/queries/coalesce_numeric_fallback.ppl [hunk 4]
```diff
@@ -0,0 +1,3 @@
+source = big5
+| eval result = coalesce(`metrics.size`, 123, 'fallback')
```
Comment (noCharger) [inline integ-test/src/test/resources/big5/queries/coalesce_numeric_fallback.ppl]: Does field type matter for `coalesce`?

**Review by noCharger — COMMENTED** [review]
Can we add field with null value to just demonstrate functionality?

File: integ-test/src/test/resources/big5/queries/coalesce_numeric_fallback.ppl (hunk 4 above)
Comment (aalva500-prog) [reply integ-test/src/test/resources/big5/queries/coalesce_numeric_fallback.ppl]: @noCharger No, we have removed strict same-type requirement for all arguments. The new implementation of the `coalesce` function performs automatic type coercion for mixed data types.

**Review by LantaoJin — APPROVED** [review]

File: integ-test/src/test/resources/big5/queries/coalesce_empty_string_priority.ppl (hunk 3 above)
Comment (aalva500-prog) [reply integ-test/src/test/resources/big5/queries/coalesce_empty_string_priority.ppl]: @noCharger Yes, you are right, this is an edge case. I just wanted to test the new functionality with `big5` data set, as the new `coalesce` function supports empty strings ("") and whitespace strings (" ") as valid values. For example, empty strings are treated as valid non-null values, not skipped. So, this query will return an empty string always. Personally, I believe this functionality will be barely used, so maybe I can remove this query.

**Comment by aalva500-prog** [comment]
@noCharger To correctly demonstrate `null` values handling, we need to verify the data in the whole `big5` data set to see if there are fields with a `null` value assigned. However, with the new implementation, we can demonstrate `null` values handling with `non-existing` fields also, as `non-existing` fields are treated as `null`. 

For example, the following query will always skip `dummy_field` field. Then, it will continue to the `host.name` field and check if it contains a `null` value. If it does contain a `null` value, it will then check if `metrics.size` is `non-null`. Finally, if both `host.name` and `metrics.size` are null, it will fallback to the `'unknown'` string literal. 

```
source = big5
| eval result = coalesce(dummy_field, `host.name`, `metrics.size`, 'unknown')
| head 10
```

Here we are demonstrating mixed data type handling in the same `coalesce` function, along with `non-existing` fields treated as `null` and string literal fallback. I think for the purpose of this exercise, we can leave this query only. Please let me know what you think, thanks!

**Comment by noCharger** [comment]
> @noCharger To correctly demonstrate `null` values handling, we need to verify the data in the whole `big5` data set to see if there are fields with a `null` value assigned. However, with the new implementation, we can demonstrate `null` values handling with `non-existing` fields also, as `non-existing` fields are treated as `null`.
> 
> For example, the following query will always skip `dummy_field` field. Then, it will continue to the `host.name` field and check if it contains a `null` value. If it does contain a `null` value, it will then check if `metrics.size` is `non-null`. Finally, if both `host.name` and `metrics.size` are null, it will fallback to the `'unknown'` string literal.
> 
> ```
> source = big5
> | eval result = coalesce(dummy_field, `host.name`, `metrics.size`, 'unknown')
> | head 10
> ```
> 
> Here we are demonstrating mixed data type handling in the same `coalesce` function, along with `non-existing` fields treated as `null` and string literal fallback. I think for the purpose of this exercise, we can leave this query only. Please let me know what you think, thanks!

Sounds good, do we still need all cases or just one query demonstrate all?

**Comment by aalva500-prog** [comment]
@noCharger IMO, one query should be enough. I'll update the PR and let you know.

**Review by noCharger — APPROVED** [review]
